### ISSUE TITLE
Contract Support

### ISSUE THREAD

**Issue by kcole16** [nearprotocol/near-explorer#204]
### Overview
Contract interactions are likely to make up a major portion (if not the majority) of the activity on NEAR. Users, both developers and end users, will want to dive into these interactions, and learn more about the contracts themselves and usage patterns.

### User Stories
As a user, I want to determine whether or not there is a contract on an account, and view any interactions with that contract, so I can know how to interact with the account.

As a user, I want to view the history of an account's deployed contracts, as well as whether or not the contract is capable of being changed in the future, so I can determine if the contract is safe to interact with.

As a technical user, I want to see the verified source code of a contract, so I can determine if the contract is safe to interact with.

As a user, I want to view a list of accounts with contracts and view their usage/popularity, so I can better understand how NEAR is used.

**Comment by bowenwang1996** [comment]
Source code of the contract is not stored on-chain.

**Comment by kcole16** [comment]
> Source code of the contract is not stored on chain.

Right, which is why contract verification will be pretty centralized, similar to [etherscan](https://etherscan.io/address/0x2af47a65da8cd66729b4209c22017d6a5c2d2400#code)

**Comment by bowenwang1996** [comment]
I see. What happens if the developer of the contract refuses to reveal the source code?

**Comment by kcole16** [comment]
It's opt in, so the default will be "unverified". Depending on how much we want to encourage devs to reveal source code, we could make the warning pretty prominent. 

Etherscan just hides the "Contract" tab when there is no verified source code, but there have been a few "public shamings" on Twitter for high profile contracts to be verified.
